Modules/Filtering/Thresholding/include/itkHistogramThresholdImageFilterBase.h

```diff
@@ -0,0 +1,202 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkHistogramThresholdImageFilterBase_h
+#define itkHistogramThresholdImageFilterBase_h
+
+#include "itkImageToImageFilter.h"
+#include "itkHistogram.h"
+
+namespace itk
+{
+
+/**
+ *\class HistogramThresholdImageFilterBase
+ * \brief Abstract base class for histogram threshold image fileters.
+ *
+ * This filter creates a binary thresholded image that separates an
+ * image into foreground and background components.
+ *
+ * The filter also has the option of providing a mask, in which case
+ * the histogram and therefore the threshold is computed from the
+ * parts of the mask with values indicated by MaskValue. The output
+ * image is, by default, masked by the same image. This output
+ * masking can be disabled using SetMaskOutput(false). Note that there
+ * is an inconsistency here. The MaskImageFilter (used internally)
+ * masks by non zero values, where as the MaskedImageToHistogramFilter
+ * uses explicit values. If this does not match your usage then the
+ * output masking will need to be managed by the user.
+ *
+ * \author Richard Beare. Department of Medicine, Monash University,
+ * Melbourne, Australia.
+ * \author Gaetan Lehmann. Biologie du Developpement et de la Reproduction, INRA de Jouy-en-Josas, France.
+ *
+ * This implementation was taken from the Insight Journal paper:
```

Comment:
"This implementation was based on the Insight Journal paper:"

It also looks like you are authoring this class :)

Reply:
Thanks for considering it Brad, but I think the PR/commit authorship is enough credit.